x/ccv/provider/types/params.go

```diff
@@ -57,24 +65,31 @@ func (p Params) Validate() error {
 	if p.TemplateClient == nil {
 		return fmt.Errorf("template client is nil")
 	}
-	if ccvtypes.ValidateDuration(p.CcvTimeoutPeriod) != nil {
-		return fmt.Errorf("ccv timeout period is invalid")
+	if err := validateTemplateClient(*p.TemplateClient); err != nil {
+		return err
+	}
+	if err := ccvtypes.ValidateDuration(p.CcvTimeoutPeriod); err != nil {
+		return err
 	}
-	return validateTemplateClient(*p.TemplateClient)
+	if err := ccvtypes.ValidatePositiveInt64(p.TrustingPeriodFraction); err != nil {
```

Comment:
`TrustingPeriodFraction` is not an integer since it's a fraction. We could use `ibctmtypes.Fraction`. See  `ibctmtypes.DefaultTrustLevel`.

Reply:
See above